gcloud-java-bigquery/src/test/java/com/google/gcloud/bigquery/CopyJobInfoTest.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.bigquery;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import com.google.common.collect.ImmutableList;
+import com.google.gcloud.bigquery.JobInfo.CreateDisposition;
+import com.google.gcloud.bigquery.JobInfo.WriteDisposition;
+
+import org.junit.Test;
+
+import java.util.List;
+
+public class CopyJobInfoTest {
+
+  private static final TableId SOURCE_TABLE = TableId.of("dataset", "sourceTable");
+  private static final List<TableId> SOURCE_TABLES = ImmutableList.of(
+      TableId.of("dataset", "sourceTable1"),
+      TableId.of("dataset", "sourceTable2")
+  );
+  private static final TableId DESTINATION_TABLE = TableId.of("dataset", "destinationTable");
+  private static final CreateDisposition CREATE_DISPOSITION = CreateDisposition.CREATE_IF_NEEDED;
+  private static final WriteDisposition WRITE_DISPOSITION = WriteDisposition.WRITE_APPEND;
+  private static final JobId JOB_ID = JobId.of("job");
+  private static final JobStatus JOB_STATUS = new JobStatus(JobStatus.State.DONE);
+  private static final JobStatistics JOB_STATISTICS = JobStatistics.builder()
+      .creationTime(1L)
+      .endTime(3L)
+      .startTime(2L)
+      .build();
+  private static final CopyJobInfo COPY_JOB =
+      CopyJobInfo.builder(DESTINATION_TABLE, SOURCE_TABLE)
```

Comment:
In at least one of the subclasses' tests, I think the super class `JobInfo` fields (i.e. `selfLink`, `etag`, etc) should be set.  Some of the fields aren't set in any of the tests, so I don't think `JobInfo`'s `builder` and `toBuilder` are getting tested fully.